mysql-async/src/main/java/com/github/jasync/sql/db/mysql/MySQLConnection.kt

```diff
@@ -340,6 +343,25 @@ class MySQLConnection @JvmOverloads constructor(
     }
 
     override fun sendQueryDirect(query: String): CompletableFuture<QueryResult> {
+        return chainQueryPromiseIfExists(query, this::send)
+    }
+
+    private fun <T> chainQueryPromiseIfExists(
+        param: T,
+        func: (T) -> CompletableFuture<QueryResult>
+    ): CompletableFuture<QueryResult> {
+        return try {
+            if (isQuerying()) {
+                queryPromise().get().flatMap { func(param) }
```

Comment:
can you please add a info log here, that we are doing a chaining?

Reply:
Thank you for your review and inspiring comments. I've tested this fix and it has some side effects as you said.
I think https://github.com/jasync-sql/jasync-sql/pull/347 is a better option.